### PR TITLE
Further Watchlist Improvements

### PR DESCRIPTION
**Pull request by AugusteBa**
### Brief Summary
Moved the watchlist scrollbar functionality in common StockFlux-Components and applied it to StockFlux-Search.

General CSS tidy-up, removed some CSS variables due to most of them being moved to the StockFlux-Components now.

### Screenshots
**Watchlist**
image   image

**Search**
image   image

### CODE REVIEW

File: stockflux-components/src/components/ScrollWrapperY/ScrollWrapperY.css [hunk 1]
```diff
@@ -0,0 +1,14 @@
+.scroll-wrapper-y {
+  flex: 1 1 auto;
+  position: relative;
+  overflow-y: hidden;
+  overflow-x: hidden;
+  -webkit-transition: 0.6s all;
+  transition: 0.6s all;
+  height: inherit;
```
Comment (oriondean) [inline stockflux-components/src/components/ScrollWrapperY/ScrollWrapperY.css]: Do we need this inherit style?

File: stockflux-components/src/components/ScrollWrapperY/ScrollWrapperY.css [hunk 2]
```diff
@@ -0,0 +1,14 @@
+.scroll-wrapper-y {
+  flex: 1 1 auto;
+  position: relative;
+  overflow-y: hidden;
+  overflow-x: hidden;
+  -webkit-transition: 0.6s all;
+  transition: 0.6s all;
```
Comment (oriondean) [inline stockflux-components/src/components/ScrollWrapperY/ScrollWrapperY.css]: Can this be more restrictive than `all`?

File: stockflux-components/src/components/ScrollWrapperY/ScrollWrapperY.css [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  height: inherit;
+}
+
+.scroll-padding-right:hover {
+  padding-right: 13px !important;
```
Comment (oriondean) [inline stockflux-components/src/components/ScrollWrapperY/ScrollWrapperY.css]: Does this need !important?

File: stockflux-components/src/components/ScrollWrapperY/ScrollWrapperY.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+  useEffect(() => {
+    checkIfOverflowing();
+    window.addEventListener("resize", checkIfOverflowing);
+    return () => {
+      window.removeEventListener("resize", checkIfOverflowing);
+    };
+  }, []);
+
+  useEffect(() => {
```
Comment (oriondean) [inline stockflux-components/src/components/ScrollWrapperY/ScrollWrapperY.js]: useEffect(checkIfOverflowing, [props.contentChangedCallback])?

File: stockflux-components/src/components/ScrollWrapperY/ScrollWrapperY.css [hunk 5]
```diff
@@ -0,0 +1,14 @@
+.scroll-wrapper-y {
```
Comment (oriondean) [inline stockflux-components/src/components/ScrollWrapperY/ScrollWrapperY.css]: Could this use CSS modules?

File: stockflux-components/src/components/ScrollWrapperY/ScrollWrapperY.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    checkIfOverflowing();
+  }, [props.contentChanged]);
+
+  const checkIfOverflowing = () => {
+    if (scrollWrapperRef.current)
+      if (
```
Comment (oriondean) [inline stockflux-components/src/components/ScrollWrapperY/ScrollWrapperY.js]: Could shorten this to
`
scrollWrapperRef.current.classList.toggle('scroll-padding-right', scrollWrapperRef.current.scrollHeight > scrollWrapperRef.current.clientHeight);
`

File: stockflux-search/src/App.js [hunk 7]
```diff
@@ -90,10 +90,20 @@ const App = () => {
             <Components.Titlebar />
             <input type="text" className={styles.input} onChange={event => setQuery(event.target.value)} placeholder="Enter stock name or symbol" />
             <div className={styles.containerList}>
-                { !isSearching && results && results.length ?
-                    results.map((result) => (<SearchResult key={result.code} code={result.code} name={result.name} />)) :
-                    <div className={styles.message}>{getMessage(searchState, results)}</div>
-                }
+                <Components.ScrollWrapperY contentChanged={Boolean(results)}>
```
Comment (oriondean) [inline stockflux-search/src/App.js]: Not sure if contentChanged will work as intended here, isn't it just going to run only once after we initially get results?

File: stockflux-watchlist/src/components/ClosedWindows/ClosedWindows.js [hunk 8]
```diff
@@ -1,7 +1,7 @@
-import React, { useState } from "react";
-import moment from "moment";
-import closedWindowsImageInactive from "../../assets/png/closed_tabs.png";
-import closedWindowsImageActive from "../../assets/png/closed_tabs_active.png";
+import React, { useState } from 'react';
+import moment from 'moment';
+import closedWindowsImageInactive from 'stockflux-components/src/images/closed_tabs.png';
```
Comment (oriondean) [inline stockflux-watchlist/src/components/ClosedWindows/ClosedWindows.js]: We should use icon classes for these rather than importing them directly from source of stockflux-components

File: stockflux-components/src/components/ScrollWrapperY/ScrollWrapperY.css (hunk 1 above)
Comment (AugusteBa) [reply stockflux-components/src/components/ScrollWrapperY/ScrollWrapperY.css]: Yes, the parent sets expected view height, which scrollbar wrapper inherits. This allows us to calculate when the overflow happens, aka when scroll height is higher than client height. If we did not inherit this, then scroll height and client height would be the same.

File: stockflux-components/src/components/ScrollWrapperY/ScrollWrapperY.css (hunk 2 above)
Comment (AugusteBa) [reply stockflux-components/src/components/ScrollWrapperY/ScrollWrapperY.css]: Yes indeed, here's the fix -> 08cff915e0c7c162c3c71c72e9adb0e02ba1de6c

File: stockflux-components/src/components/ScrollWrapperY/ScrollWrapperY.css (hunk 3 above)
Comment (AugusteBa) [reply stockflux-components/src/components/ScrollWrapperY/ScrollWrapperY.css]: Needed to be at some point, but not anymore, thanks for noticing! Fixed -> 08cff915e0c7c162c3c71c72e9adb0e02ba1de6c

File: stockflux-components/src/components/ScrollWrapperY/ScrollWrapperY.js (hunk 4 above)
Comment (AugusteBa) [reply stockflux-components/src/components/ScrollWrapperY/ScrollWrapperY.js]: Indeed, much nicer, fixed ->08cff915e0c7c162c3c71c72e9adb0e02ba1de6c

File: stockflux-components/src/components/ScrollWrapperY/ScrollWrapperY.css (hunk 5 above)
Comment (AugusteBa) [reply stockflux-components/src/components/ScrollWrapperY/ScrollWrapperY.css]: As discussed today, not really as stockflux-components were not created using create-react-app.

File: stockflux-components/src/components/ScrollWrapperY/ScrollWrapperY.js (hunk 6 above)
Comment (AugusteBa) [reply stockflux-components/src/components/ScrollWrapperY/ScrollWrapperY.js]: That's neat! Thanks - > 08cff915e0c7c162c3c71c72e9adb0e02ba1de6c

File: stockflux-search/src/App.js (hunk 7 above)
Comment (AugusteBa) [reply stockflux-search/src/App.js]: You're right, just tested. Removed the boolean, so every time when we get results the prop would update and trigger overflow recalculation. Fixed -> 08cff915e0c7c162c3c71c72e9adb0e02ba1de6c

File: stockflux-watchlist/src/components/ClosedWindows/ClosedWindows.js (hunk 8 above)
Comment (AugusteBa) [reply stockflux-watchlist/src/components/ClosedWindows/ClosedWindows.js]: Fixed -> 08cff915e0c7c162c3c71c72e9adb0e02ba1de6c
